audits/supersimple.py

```diff
@@ -0,0 +1,175 @@
+                            ...
+                        },
+                        ...
+                    }
+        """
+
+        rho = -math.log(self.risk_limit)/((1/(2*self.gamma)) + self.l*math.log(1 - 1/(2*self.gamma))) 
+
+        diluted_margin = self.compute_diluted_margin(contests, margins, total_ballots)
+
+        return math.ceil(rho/diluted_margin) 
+
+    def compute_risk(self, contests, margins, cvrs, sample_cvr):
+        """
+        Computes the risk-value of <sample_results> based on results in <contest>.
+
+        Inputs: 
+            contests       - the contests and results being audited
+            margins        - the margins for the contest being audited
+            cvrs           - mapping of ballot_id to votes:
+                    {
+                        'ballot_id': {
+                            'contest': {
+                                'candidate1': 1,
+                                'candidate2': 0,
+                                ...
+                            }
+                        ...
+                    }
+
+            sample_cvr - the CVR of the audited ballot  
+                    {
+                        'ballot_id': {
+                            'contest': {
+                                'candidate1': 1,
+                                'candidate2': 0,
+                                ...
+                            }
+                        ...
+                    }
+
+        Outputs:
+            measurements    - the p-value of the hypotheses that the election
+                              result is correct based on the sample, for each winner-loser pair. 
+            confirmed       - a boolean indicating whether the audit can stop
+        """
+        
+        p = 1
+
+
+        diluted_margin = self.compute_diluted_margin(contests, margins, len(cvrs))
+        V = diluted_margin*len(cvrs)
+
+        U = 2*self.gamma/diluted_margin
+
+        lowest_p = 1
+        result = False
+        for ballot in sample_cvr:
+            e_r = 0
+
+            for contest in contests:
+                if contest not in sample_cvr[ballot]:
+                    # TODO I think this is right
+                    continue
+                for winner in margins[contest]['winners']:
+                    for loser in margins[contest]['losers']:
+                        v_w = cvrs[ballot][contest][winner]
+                        a_w = sample_cvr[ballot][contest][winner]
+
+                        v_l = cvrs[ballot][contest][loser]
+                        a_l = sample_cvr[ballot][contest][loser]
+
+                        V_wl = contests[contest][winner] - contests[contest][loser]
+
+                        e = ((v_w - a_w) - (v_l - a_l))/V_wl
+                        if e > e_r:
+                            e_r = e
+
+            p_b = (1 - 1/U)/(1 - (e_r/((2*self.gamma)/V)))
+            p *= p_b
+
+            if p < lowest_p:
```

Comment:
This is mathematically correct if the CVRs are guaranteed to be in unsorted sample order -- but as a matter of policy, I would be predisposed to use all available audit data. It's uncomfortable if attaining the risk limit depends on whether an observed overstatement is or isn't included in the calculation.

Reply:
Ah, I had to add this in because of the way the SuperSimple paper presented its data. I'm going to rework it all with numbers generated from [Stark's tool](https://www.stat.berkeley.edu/~stark/Vote/auditTools.htm) and go from there.